add agent_dvr (#12333)
home-assistant/home-assistant.github.io: source/_integrations/agent_dvr.markdown

---
title: Agent DVR
description: Access and control Agent DVR from Home Assistant.

> logo: agent_dvr.png

Reviewer: Please open up a PR to our brands repository to add the logo & icon of the brand, and remove the `logo` directive from this PR.

<https://github.com/home-assistant/brands>

Be sure to add link between the 2 PR's in the openings post of both.

ha_category:
  - Camera
  - Binary Sensor
  - Alarm Control Panel
ha_config_flow: true
ha_release: 0.107
ha_iot_class: Local Pull
ha_codeowners:
  - '@ispysoftware'
ha_domain: agent_dvr
---

[Agent DVR](https://www.ispyconnect.com/download.aspx/) is a free* DVR software solution for windows 10. Agent DVR runs as a service and can access and control a huge range of third party cameras.

> Home Assistant will automatically discover Agent DVR on your network or you can add it manually via integrations using the IP address and Port of the server eg: http://192.168.1.3:8090/

Reviewer: ```suggestion
Home Assistant will automatically discover Agent DVR on your network or you can add it manually via integrations using the IP address and port of the server e.g.,: `http://192.168.1.3:8090/`.
```

## Configuration

> For configuration go to the `Integrations pane` on your Home Assistant instance.

Reviewer: ```suggestion
For configuration, go to the `Integrations panel` on your Home Assistant instance.
```

<div class='note'>
  Please ensure you are using Agent DVR v2.6.1.0 +
</div>

## Binary Sensor

The following sensor types are supported:

- Motion detection (Agent Alerts)
- Camera Online/ Offline state

## Alarm Control Panel

The alarm control panel can enable or disable the master Alert functions in Agent and set the profile to Home, Away or Night. You can configure the profile (which cameras are enabled/ recording) within Agent from the server menu.

> ## Services

Reviewer: Please list the services and their configuration options.

- Enable/ Disable individual devices
- Record/ Stop Record individual devices
- Enable/ Disable alerts on individual devices

## IFrame

>  - Using the iframe integration you can embed the Agent viewer directly in Home Assistant. Just point it to https://www.ispyconnect.com/app/

Reviewer: I think you mean the iframe Lovelace card, not integration?
Please provide links and examples for that.

Furthermore, the iframe Lovelace card as been renamed in 0.107

---
title: Agent DVR
description: Access and control Agent DVR from Home Assistant.
ha_category:
  - Camera
ha_config_flow: true
ha_release: 0.108.6
ha_iot_class: Local Pull
ha_codeowners:
  - '@ispysoftware'
ha_domain: agent_dvr
---

> [Agent DVR](https://www.ispyconnect.com/download.aspx/) is a free* software DVR solution for windows 10, Mac and Linux. Agent DVR runs as a service or console application and can access and control a huge range of third party cameras with advanced motion detection including YOLO integration for object recognition. The iSpyConnect website provides secured (SSL) remote access without port forwarding needed under a subscription model.
> 
> You can add Agent DVR via integrations using the IP address and port of the server e.g.,: `http://192.168.1.3:8090/`.
> 

Reviewer: ```suggestion
[Agent DVR](https://www.ispyconnect.com/download.aspx/) is a free* software DVR solution for windows 10, Mac and Linux. Agent DVR runs as a service or console application and can access and control a huge range of third party cameras with advanced motion detection, including YOLO integration for object recognition. The iSpyConnect website provides secured (SSL) remote access without port forwarding needed under a subscription model.

You can add Agent DVR via integrations using the IP address and port of the server, e.g.,: `http://192.168.1.3:8090/`.

```

Author: sorry, i can't see what you changed here?

Reviewer: added a comma here:
`advanced motion detection, including`

and here:
`of the server, e.g.,:`

Added a new line after this section.

## Configuration

> For configuration, go to the `Integrations panel` on your Home Assistant instance.

Reviewer: What should one do after he went to the integrations panel?

Please don't code fence "Integrations panel" as it is not code.

Author: i've never managed to get anything working in hass using configuration.yaml and have been unable to test anything with it - would you be able to help me out here? I've only ever used the integrations menu/ UI.

Reviewer: I meant; if one goes to the integration page... one would be looking at all the integrations they have installed. What is next?

Maybe explain the user to click the + sign and pick "X" from the list and follow the instructions on screen?

<div class='note'>

>   Please ensure you are using Agent DVR v2.6.1.0 +

Reviewer: ```suggestion
  Please ensure you are using Agent DVR v2.6.1.0+
```

</div>

## Services

Once loaded, the `agent_dvr` integration will expose services that can be called to perform various actions. The `entity_id` service attribute can specify one or more specific cameras.

Available services:
`enable_alerts`, `disable_alerts`,
`start_recording`, `stop_recording`,
`turn_on`, `turn_off`, `toggle`, `enable_motion_detection`,`disable_motion_detection`

#### Service `enable_alerts`/`disable_alerts`

These services enable or disable the device's alert events within Agent DVR.

Service data attribute | Optional | Description
-|-|-

> `entity_id` | no | Name(s) of entities, e.g., `camera.living_room_camera`.

Reviewer: `entity_id` is optional, because one could be targeting an area alternatively.

#### Service `start_recording`/`stop_recording`

These services start or stop the device recording.

Service data attribute | Optional | Description
-|-|-
`entity_id` | no | Name(s) of entities, e.g., `camera.living_room_camera`.

#### Service `turn_on`/`turn_off`/`toggle`

These services turn on, off or toggle the device enabled state within Agent DVR

Service data attribute | Optional | Description
-|-|-

> `entity_id` | no | Name(s) of entities, e.g., `camera.living_room_camera`.

Reviewer: These services are inherited from the `camera` component and should not be documented here.

> ## IFrame

Reviewer: Maybe it to: Embedding into the Home Assistant frontend or something?
IFrame might not be a term the more novice people are familiar with.
